### PR TITLE
support use of slackbot as a library

### PR DESCRIPTION
**Pull request by 9len**
We vendor (and love) slackbot, but we don't like having to regenerate init.go everytime we update our vendored code. This change adds an alternate method of starting up slackbot that allows people to construct the `map[string][]robots.Robot` themselves and pass it into a `server.Main` function from within their own `main` function. 

I've tested this approach with our code, taking care to make sure I haven't disturbed the existing way of doing things. I did this by first making the change to slackbot and leaving our code as-is. Once that worked, I moved our code onto the new way, which also worked.

I added example usage to the README, and a snippit.

One other minor change: the `getRobots` method was unnecessary; getting non-existing values for from a map will return the empty value, which for a slice is `nil`, so there's no need to check `ok` and return `nil` by hand. I replaced calls to this method with `robots.Robots[robot.command]`.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,134 +1,11 @@
...
-	"log"
-	"net/http"
-	"os"
-	"strings"
-
-	"github.com/trinchan/slackbot/Godeps/_workspace/src/github.com/gorilla/schema"
-	_ "github.com/trinchan/slackbot/importer"
-	"github.com/trinchan/slackbot/robots"
+	_ "github.com/9len/slackbot/importer"
+	"github.com/9len/slackbot/robots"
```
Comment (trinchan) [inline main.go]: Please update these imports to be for this repo.

**Comment by trinchan** [comment]
Looks good! I'm for the idea.

File: main.go (hunk 1 above)
Comment (9len) [inline main.go]: Whoops! did a big search/replace for these, but main.go didn't make it into the commit. rebased/pushed now. should be all set.

**Comment by trinchan** [comment]
thanks for your contribution!

**Pull request merged**